geospaas_processing/copiers.py

```diff
@@ -42,31 +43,44 @@ def write_flag_file(type_in_flag_file, source_path, dataset, destination_filenam
             flag_file.write(string_to_write)
 
     def file_or_symlink_copy(self, source_paths, dataset):
-        """copy the file or a symlink of the file of dataset based on its stored local address in
-        the database."""
+        """copy the file or a symlink of the file/folder of dataset based on its stored local
+        address in the database."""
         for source_path in source_paths:
-            if os.path.isfile(source_path.uri):
-                destination_filename = os.path.join(
-                    self._destination_path, os.path.basename(source_path.uri))
+            self.source_is_file = os.path.isfile(source_path.uri)
+            self.source_is_folder = os.path.isdir(source_path.uri)
```

Comment:
Please don't use object attributes like pseudo global variables.
For this usage, it's much safer to pass the information as arguments to the `copy_item()` method, because if `copy_item()` is called from somewhere else than `file_or_symlink_copy()`, it will rely on the wrong information (`self.source_is_file()` and `self.source_is_folder()` will have the values of the last call to `file_or_symlink_copy()`).

Alternatively, just use `os.path.exists()` in `file_or_symlink_copy()`, and call `os.path.isfile()` and `os.path.isdir()` directly in `copy_item()`

Reply:
Done in the next commit.